### ISSUE TITLE
New store dashboard / stats at a glance.

### ISSUE THREAD

**Issue by danieliser** [easydigitaldownloads/easy-digital-downloads#4714]
I would like to propose a new Admin page in the EDD Downloads Menu. Available only to store owners, managers etc.

It would essentially be a go to page for most stores where they can get all of the useful stats at a glance. Think the EDD iOS app but in the dashboard with nice pretty charts and such.

I have a lot of this done as a front end tool, but think it would be appropriate for the core. Every store would benefit from a store dashboard.

I would be happy to get this going in a new branch and with your oversight, and then make it extendable so that you can build on it for the future as well.

**Comment by cklosowski** [comment]
I like the idea of this. Wondering if it would be served better as a free or cheap add on? Thoughts @easydigitaldownloads/core-devs ?

**Comment by cklosowski** [comment]
On the flip side of it's built into EDD core we can give a good example of how to build on top of it for other extensions.

**Comment by danieliser** [comment]
@cklosowski Correct, but also consider that nearly all other eCommerce platforms, especially those outside WP have a unified dashboard. I think adding this is one step closer to being the complete package.

We all know that reporting is one of the most crucial parts of a store for the owners, EDD reporting is getting better, especially once it moves payments to a separate table, but this would be a better place for the average person to start their day.

The other reports are less overview and more specific already, so all that is really missing is the one place to see it all, then have links that take them to each separate report etc. Similar to a google analytics dash.

**Comment by danieliser** [comment]
Here is a public page that I go to for a quick glance, but I would add a lot more info to a private page obviously.

https://wppopupmaker.com/open-stats/

The sales stream and stats can easily be set up to update using AJAX polling etc.

**Comment by chriscct7** [comment]
For reference Jigoshop 2.0 beta has this as a top level page

**Comment by cklosowski** [comment]
Yeah, one thing that'll have to be done first is moving all our payment data into a custom table. We simply don't have a good infrastructure for a good reporting platform as of yet. I'll milestone this for a bit down the road.

**Comment by cklosowski** [comment]
The reason I say this is that building a dashboard is going to be a fairly intensive process for reporting purposes and it has the potential to be pretty harsh on the DB.

**Comment by danieliser** [comment]
Maybe later, but currently due to limitations of the current data structure, the resources needed are minimal. Currently though reporting on basic stats that are already available would serve a purpose, both to give users a better overview of the store, but also as a starting point for the next wave of reporting.

**Comment by SDavisMedia** [comment]
I do personally like this idea and for it to be in core, not an extension.

**Comment by cklosowski** [comment]
That's what I'm stating, our stats right now hammer the database which is why they are heavily cached. I don't feel like this should be included until we can get #4678 into something that's reusable so that each graph or data point on the dashboard can be built to be AJAX'd in.

**Comment by cklosowski** [comment]
It's milestone'd, so we can work on this as we get moving forward. `2.7` is pretty full as it is, so `2.8` is the next major milestone we'll shoot for to include this.

**Comment by danieliser** [comment]
Sorry, let me outline a few things: All of the data I am talking about displaying quickly (now, since other data is such a PITA to compile) are stored in site options or require a simple post query. Eventually more stats would be possible, but for now most of what I and most admins need is already available in a highly available format with no overhead resources needed.

**Comment by danieliser** [comment]
But I am happy either way to see it milestone'd :).

**Comment by cklosowski** [comment]
@danieliser Perfect. That's a good thing, the milestone just means that we've got it on the roadmap. The goal is to really speed up core release cycles, so hopefully 2.8 isn't 12 months away, but more like 6

**Comment by danieliser** [comment]
Maybe a feature plugin in the mean time similar to WP core would suffice. Would serve as a place for feedback from the community so you can kind of feel it out before committing a ton of time to it in core.

It can include the top level dash page, all currently available stats, and likely make it widgetable like the WP Core dash.

**Comment by cklosowski** [comment]
If we're going to do this, I think it might be a good idea to look at mimicing what the WP Dashboard does (to allow for minimal user training). Register an EDD Dashboard widget, and allow them to be moved and placed where needed. This is beneficial as users already know how to use it, it's already mobile responsive, and it's slick.

**Comment by cklosowski** [comment]
@danieliser :laugh:

**Comment by danieliser** [comment]
Lol @cklosowski Great minds...

I was going to suggest also that we could concept it in the WP Core dash. IE just make a slew of widgets that can be inserted on each users dash, so they can play with the idea on their own now.

**Comment by danieliser** [comment]
The only thing it will need that WP Core doesn't already include is a settings icon for each widget that allows customizing each one (if customization is available), such as date ranges, filters etc. 

Should be simple enough though to create an extendable dropdown for each widget **in the future**.

**Comment by sunnyratilal** [comment]
I really like the idea of a unified dashboard but agree with @cklosowski in that we definitely need to fix up our database structure first so that the dashboard would load fairly quickly and be efficient when querying the database too.

**Comment by paranoidddd** [comment]
Why not use the API or create a plugin for this?  Sounds like a cool idea but going into core just adds a layer of bloat.  Stuff like this may push me and my clients away from doing so in the future.  I or we may or may not be a big spender on EDD products.

It is not needed to run your store and will be CPU intensive for bigger stores.  EDD is adding too much fat lately into the plugin.


NOT EVERY USER WANTS 1000000 FEATURES.

**Comment by michaelbeil** [comment]
I like the idea of adding a dashboard once the infrastructure is built out.  Would be fascinating to do a feature plugin for a later merge in EDD core.

**Comment by shazahm1** [comment]
Looks like a plugin is available which adds this. Personally, I would love something like this to be included in core as a reports/metric dashboard.

https://wordpress.org/plugins/edd-metrics/

**Comment by JJJ** [comment]
I think the new Reports interface will satisfy a lot of the desire here.

Moving this to 3.1 to let Reports launch, see if it's enough, and revisit. We can either work on a Dashboard, or emphasize Reports more, or maybe we'll all be happy with Reports as-is.

**Comment by cklosowski** [comment]
I'm going to remove the milestone, not because I don't want to do this but I need to spend some time using the new Reports in 3.0 so I can work through the UI/UX of it all. I'm liking this idea more and more though.
